praw/reddit.py

```diff
@@ -648,9 +653,20 @@ def request(
             (default: None).
 
         """
-        return self._core.request(
-            method, path, data=data, files=files, params=params
-        )
+        try:
+            return self._core.request(
+                method, path, data=data, files=files, params=params
+            )
+        except BadRequest as exception:
+            data = exception.response.json()
```

Comment:
This assumes that there is always JSON data. Try and catch for bad json, like in PRAWCore.

Reply:
I added an new explicit exception so we can learn about cases where a BadRequest may not have a json body (if that ever happens).